File: spring/src/main/java/ru/dfsystems/spring/tutorial/dto/student/StudentDto.java
package ru.dfsystems.spring.tutorial.dto.student;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;
import ru.dfsystems.spring.tutorial.dto.BaseDto;
import ru.dfsystems.spring.tutorial.dto.course.CourseList;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
public class StudentDto extends BaseDto<StudentHistory>{
    private String firstName;
    private String lastName;
    private String middleName;
    private String passport;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate birthDate;

    private List<CourseList> courses;
}

File: angular/src/app/_model/student/studentDto.ts
import {BaseEntity} from "../base/base-entity";
import {StudentHistory} from "./student-history";
import {CourseList} from "../../../../../../../../../../Study/spring-angular-project/angular/src/app/_model/course/courseList";

export class Student extends BaseEntity<StudentHistory>{
  firstName:string;
  middleName:string;
  lastName:string;
  passport:string;
  birthDate:string;
  courses:CourseList[];
  getName() {
    return "student";
  }
}
